### PR TITLE
DataGrid.Pdf - Split Grid by columns

### PR DESCRIPTION
**Pull request by KuznetsovVN**

### CODE REVIEW

File: js/exporter/jspdf/export_data_grid_2.js [hunk 1]
```diff
@@ -1,90 +1,155 @@
...
+    addRow(cells) {
+        if(!isDefined(cells)) {
+            throw 'cells is required';
+        }
+        this.rows.push(cells);
+        for(let i = 0; i < cells.length; i++) {
+            const currentCell = cells[i];
+            if(currentCell.drawLeftBorder === false) {
+                if(i >= 1) {
+                    cells[i - 1].drawRightBorder = 0;
```
Comment (IgnatovDan) [inline js/exporter/jspdf/export_data_grid_2.js]: cells[i - 1].drawRightBorder = **false**; ?

File: js/exporter/jspdf/export_data_grid_2.js [hunk 2]
```diff
@@ -1,90 +1,155 @@
...
+    }
+
+    drawTo(doc) {
+        drawTable(doc, this);
+    }
+}
+
+class PdfGrid {
+    constructor(splitByColumns) {
+        this._splitByColumns = splitByColumns ?? [];
```
Comment (IgnatovDan) [inline js/exporter/jspdf/export_data_grid_2.js:46]: imho we can remove splitByColumns from ctor and keep in startNewTable only

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/exporter/jspdf/export_data_grid_2.js [hunk 3]
```diff
@@ -1,90 +1,155 @@
...
+        this._newPageTables = [];
+        this._tables = [];
+        this._currentHorizontalTables;
```
Comment (IgnatovDan) [inline js/exporter/jspdf/export_data_grid_2.js]: 'no effect' line

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/exporter/jspdf/export_data_grid_2.js (hunk 1 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/export_data_grid_2.js]: fixed

File: js/exporter/jspdf/export_data_grid_2.js (hunk 3 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/export_data_grid_2.js]: Fixed

File: js/exporter/jspdf/export_data_grid_2.js (hunk 2 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/export_data_grid_2.js:46]: This '_splitByColumns' property is used in two methods: 'startNewTable' and 'addRow', at the moment it is not clear how to replace it in the 'addRow' method.

**Review by IgnatovDan — APPROVED** [review]

**Pull request merged**